implementations/rust/router/src/router.rs

```diff
@@ -0,0 +1,168 @@
+#![allow(unused)]
+pub mod router {
+    use ockam_message::message::*;
+    use std::collections::HashMap;
+    use std::io::{Error, ErrorKind, Read, Write};
+    use std::net::UdpSocket;
+    use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+    use std::ops::Add;
+    use std::sync::{Arc, Mutex};
+    use std::thread;
+
+    pub struct UdpAddressHandler {
+        pub socket: UdpSocket,
+    }
+
+    impl HandleMessage for UdpAddressHandler {
+        fn handle_message(&self, mut m: Box<Message>) -> Result<(), std::io::Error> {
+            println!("In UdpAddressHandler!");
+            Ok(())
+        }
+    }
+
+    pub trait HandleMessage {
+        fn handle_message(&self, m: Box<Message>) -> Result<(), std::io::Error>;
+    }
+
+    pub struct Router {
+        registry: HashMap<u64, Arc<Mutex<Box<dyn HandleMessage + Send>>>>,
+    }
+
+    impl Router {
+        pub fn new() -> Router {
+            Router {
+                registry: HashMap::new(),
+            }
+        }
+
+        pub fn register_handler(
+            &mut self,
+            handler: Arc<Mutex<Box<dyn HandleMessage + Send>>>,
+            address: &Address,
+        ) -> Result<(), String> {
```

Comment:
This function, I think, should set a handler by `address type` not by `address`.
The signature in the elixir implementation is:

```
set_message_handler_for_address_type(address_type, handler)
```

Reply:
Ok, it was an incomplete refactor after the design changes from a couple days ago.